### ISSUE TITLE
Additional dependencies and rule for `make regen-all`

### ISSUE THREAD

**Issue by serhiy-storchaka** [python/cpython#74823]
BPO | [30638](https://bugs.python.org/issue30638)
--- | :---
Nosy | @ncoghlan, @vstinner, @zware, @serhiy-storchaka, @miss-islington
PRs | <li>python/cpython#5671</li><li>python/cpython#5679</li><li>python/cpython#5724</li><li>python/cpython#5725</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2017-06-12.13:43:01.478>
labels = ['type-feature', '3.7', 'build']
title = 'Additional dependencies and rule for `make regen-all`'
updated_at = <Date 2018-02-18.01:27:25.464>
user = 'https://github.com/serhiy-storchaka'
```

bugs.python.org fields:
```python
activity = <Date 2018-02-18.01:27:25.464>
actor = 'zach.ware'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Build']
creation = <Date 2017-06-12.13:43:01.478>
creator = 'serhiy.storchaka'
dependencies = []
files = []
hgrepos = []
issue_num = 30638
keywords = ['patch']
message_count = 6.0
messages = ['295781', '295787', '295795', '312289', '312291', '312293']
nosy_count = 5.0
nosy_names = ['ncoghlan', 'vstinner', 'zach.ware', 'serhiy.storchaka', 'miss-islington']
pr_nums = ['5671', '5679', '5724', '5725']
priority = 'normal'
resolution = None
stage = 'patch review'
status = 'open'
superseder = None
type = 'enhancement'
url = 'https://bugs.python.org/issue30638'
versions = ['Python 3.7']
```

</p></details>

**Comment by serhiy-storchaka** [comment]
`make regen-all` regenerates many generated files. But not all.

1. Argument Clinic files has their own make target, `make clinic`. Perhaps "clinic" should be a dependency for "regen-all".

2. [Lib/token.py](https://github.com/python/cpython/blob/main/Lib/token.py) and [Lib/symbols.py](https://github.com/python/cpython/blob/main/Lib/symbols.py) are generated from [Include/token.h](https://github.com/python/cpython/blob/main/Include/token.h) and [Include/graminit.h](https://github.com/python/cpython/blob/main/Include/graminit.h).

3. [Lib/keyword.py](https://github.com/python/cpython/blob/main/Lib/keyword.py) is generated from graminit.c.

4. [Modules/sre_constants.h](https://github.com/python/cpython/blob/main/Modules/sre_constants.h) is generated from [Lib/sre_constants.py](https://github.com/python/cpython/blob/main/Lib/sre_constants.py).

5. [Lib/stringprep.py](https://github.com/python/cpython/blob/main/Lib/stringprep.py) is generated by mkstringprep.py (currently it is slightly outdated).

6. [Objects/unicodetype_db.h](https://github.com/python/cpython/blob/main/Objects/unicodetype_db.h), [Objects/unicodetype_db.h](https://github.com/python/cpython/blob/main/Objects/unicodetype_db.h) and [Modules/unicodename_db.h](https://github.com/python/cpython/blob/main/Modules/unicodename_db.h) are generated by makeunicodedata.py (it downloads Unicode data from Internet).

Some generating scripts support Python 2.7 and even older versions. Some generating scripts (in particular Argument Clinic) need recent Python version (3.5+ or like).

**Comment by zware** [comment]
Agreed with all of the above, with the possible exception of 6; I don't think that we want `make regen-all` to require internet access.

Please make sure that the PR for this includes removing the explicit `make clinic` from .travis.yml.

**Comment by serhiy-storchaka** [comment]
> I don't think that we want `make regen-all` to require internet access.

Agree. I mentioned it only because these files also are generated.

**Comment by serhiy-storchaka** [comment]
Most of this is done: `regen-all` now includes clinic (GH-5671), `regen-token`, `regen-keyword` and `regen-sre`, `Lib/symbol.py` was removed in 3.10, and `regen-unicodedata` is intentionally left manual because it needs the network.

What is left is minor: `regen-re` (`Lib/re/_casefix.py`) is not included in `regen-all`, and `Lib/stringprep.py` has no target because `Tools/unicode/mkstringprep.py` reads `rfc3454.txt` from the current directory and writes to stdout. These can be filed separately if needed.
